### PR TITLE
Sema: Kick off ActorIsolationRequest after checking conformances to avoid cycles

### PR DESCRIPTION
**Pull request by slavapestov**
Fixes rdar://problem/80107082.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/Sema/TypeCheckProtocol.cpp [hunk 1]
```diff
@@ -5936,6 +5938,13 @@ void TypeChecker::checkConformancesInContext(IterableDeclContext *idc) {
   // Check all conformances.
   groupChecker.checkAllConformances();
 
+  // Check actor isolation.
+  for (auto *member : idc->getMembers()) {
+    if (auto *valueDecl = dyn_cast<ValueDecl>(member)) {
+      (void)getActorIsolation(valueDecl);
```
Comment (DougGregor) [inline lib/Sema/TypeCheckProtocol.cpp:5944]: This isn't checking the actor isolation of top-level declarations, but we should.

Comment (slavapestov) [reply lib/Sema/TypeCheckProtocol.cpp:5942]: Nice catch, fixed.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**